massa-network/src/peer_info_database.rs

```diff
@@ -494,50 +451,27 @@ impl PeerInfoDatabase {
         if !ip.is_global() {
             return Err(NetworkError::InvalidIpError(*ip));
         }
-        let (available_bootstrap_conns, available_nonbootstrap_conns) =
-            self.get_available_out_connection_attempts();
+
+        let peer_type = self.get_peer_type(ip).ok_or({
+            NetworkError::PeerConnectionError(NetworkConnectionErrorType::PeerInfoNotFoundError(
+                *ip,
+            ))
+        })?;
+
+        self.update_global_active_out_connection_attempt_count(
+            peer_type,
+            true,
+            NetworkConnectionErrorType::TooManyConnectionAttempt(*ip),
```

Comment:
toomanyconnectionattemptS (plural)

granted, the previous version was even more wrong xD